pkg/core/native/oracle.go

```diff
@@ -402,6 +409,10 @@ func (o *Oracle) verify(ic *interop.Context, _ []stackitem.Item) stackitem.Item
 	return stackitem.NewBool(ic.Tx.HasAttribute(transaction.OracleResponseT))
 }
 
+func (o *Oracle) onPayment(_ *interop.Context, _ []stackitem.Item) stackitem.Item {
+	return stackitem.Null{}
```

Comment:
Just a question, don't we need to check that calling script hash is GAS and panic if not? Otherwise, Oracle contract will accept any other token.

Reply:
I think this needs to be discussed with core devs.

Reply:
neo-project/neo#2093

Reply:
Fixed, allow GAS only.

Reply:
There is a problem with this: when invoking native methods directly, calling script hash is not updated. And when invoking them via `contract.Call`, GAS is spent.

Reply:
Same problem in neo-project/neo#2094.